app/controllers/settings/two_factor_authentication/otp_authentication_controller.rb

```diff
@@ -5,14 +5,12 @@ module TwoFactorAuthentication
     class OtpAuthenticationController < BaseController
       include ChallengableConcern
 
-      layout 'admin'
+      skip_before_action :require_functional!
 
-      before_action :authenticate_user!
+      before_action :require_not_suspended!
```

Comment:
As I said in my previous comment, I think it's probably fine to allow changing 2FA settings for suspended users.